File: task-03/sub-task 3/Java/code.java
import java.util.Scanner;

public class DiamondPattern {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter an odd number of rows for the diamond: ");
        int n = scanner.nextInt();
        
        // Upper part including the middle row
        for (int i = 0; i <= n / 2; i++) {
            System.out.println(" ".repeat((n / 2) - i) + "*".repeat(2 * i + 1));
        }

        // Lower part (excluding the middle row)
        for (int i = (n / 2) - 1; i >= 0; i--) {
            System.out.println(" ".repeat((n / 2) - i) + "*".repeat(2 * i + 1));
        }
    }
}

File: task-03/sub-task 3/Rust/code.rs
use std::io;

fn main() {
    let mut input = String::new();
    println!("Enter an odd number of rows for the diamond: ");
    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    // Upper part including the middle row
    for i in 0..=(n / 2) {
        println!(
            "{}{}",
            " ".repeat((n / 2) - i),
            "*".repeat(2 * i + 1)
        );
    }

    // Lower part (excluding the middle row)
    for i in (0..(n / 2)).rev() {
        println!(
            "{}{}",
            " ".repeat((n / 2) - i),
            "*".repeat(2 * i + 1)
        );
    }
}
